controllers/vrg_volrep.go

```diff
@@ -319,14 +319,13 @@ func (v *VRGInstance) preparePVCForVRDeletion(pvc *corev1.PersistentVolumeClaim,
 	// For Sync mode, we don't want to set the retention policy to delete as
 	// both the primary and the secondary VRG map to the same volume. The only
 	// state where a delete retention policy is required for the sync mode is
-	// when the VRG is primary. Furthermore, we need to clear the PV claimRef
-	// in order for the PV to go to the Available status phase.
+	// when the VRG is primary.
+	// Further, the PV will go to Available state, when the workload is moved back
+	// to the current cluster, in case the PVC has been deleted (cases like STS the
+	// PVC may not be deleted). This is achieved by clearing the required claim ref.
+	// such that the PV can bind back to a recreated PVC. func ref.: updateExistingPVForSync
 	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
-		v.instance.Spec.Sync != nil {
-		if err := v.clearPVClaimRefMembers(pvc); err != nil {
-			return err
-		}
-	} else {
```

Comment:
How the heck did we miss this in the review? Go figure...
Removing the block above [325-329] causes the PV to not change the reclaim policy to `Delete` when the application is deleted.